### ISSUE TITLE
wifi don't releasing

### ISSUE THREAD

**Issue by bugge95** [Android-L-Porting-Team/Android-L-Mako#84]

**Comment by rskyline** [comment]
I can confirm this issue. If you leave the area of the WiFi, no matter how far you are, it will show as connected. Leaving your phone without internet till you turn off WiFi manually.

**Comment by S1lv10Fr4gn4n1** [comment]
I have the same problem on beta4, in the beta3-pre3 this problem didn't happen.

**Comment by yvrRome** [comment]
I also confirm this issue on Beta 4.

**Comment by ruikaka16** [comment]
Yes,I have the same issue on beta4

**Comment by neoproducciones** [comment]
Same problem here. It happened to me from Beta 3 prev. 5 (The first I tried). I am connected to a Wireless N network.

**Comment by smartb0y** [comment]
Its strange for me actually.
I started with Beta3-Preview5.
I never even observed it once on Beta3-Preview5.
Yesterday I updated to Beta4.
And now, I have faced this issue multiple times !!!
Reverted back to Beta3-Preview5

**Comment by iazone** [comment]
+1

**Comment by Hamabama** [comment]
+1

**Comment by spupuz** [comment]
+1

**Comment by Sykopompos** [comment]
The Mac address and WiFi is a symptom of the new kernel. Will have to pull the old kernel an make a any kernel zip and add it to the op instead of having people revert to old versions

**Comment by inkrement** [comment]
+1

**Comment by difelice** [comment]
+1

**Comment by michaelschwarz-dev** [comment]
+1

**Comment by JohnPeel** [comment]
+1

**Comment by sagaceilo** [comment]
+1

**Comment by Sykopompos** [comment]
There is a beta 4 with the old kernel and drivers in the op right now for people instead of having them reverting to 3.5 an loosing the other fixes that 4 brought.

I just put out a test build on XDA http://forum.xda-developers.com/showthread.php?p=54251226 for trying to fix WiFi issues

**Comment by rskyline** [comment]
Thanks!

**Comment by tomburnes** [comment]
+1

**Comment by seliqui** [comment]
Same here - Problem showed up since switching beta3 to beta4. Currenty installed the nightly build from last night

**Comment by ajayclk** [comment]
Same issue on beta 4.9.1

**Comment by Hamabama** [comment]
For me 4.9.1 fixed the issue. Did you guys wiped cash before and after the installation?

**Comment by ajayclk** [comment]
Yep for me it is working fine on 4.9.1..

**Comment by iazone** [comment]
Confirm work fine +1

**Comment by michel-slm** [comment]
Yeah, this issue could be closed. The one problem I have now is that I'd sometimes lose cellular connectivity (presumably when getting to an area with poor coverage), and unlike stable versions of Android, this port doesn't automatically re-acquire it. Quickest fix is to go to airline mode and back.
